### PR TITLE
Users/nmalkapuram/integrating postgresql

### PR DESCRIPTION
**Pull request by nmalkapuram**
Integrating Postgresql

Contents : 
1. Dependencies and unit tests
2. Executors, unit tests and functional tests
3. Parser and unit tests

### CODE REVIEW

File: src/VirtualClient/VirtualClient.Actions/PostgreSQL/PostgreSQLExecutor.cs [hunk 1]
```diff
@@ -0,0 +1,475 @@
...
+                            ErrorReason.LinuxDistributionNotSupported);
+                }
+            }
+        }
+    }
+
+    /// <summary>
+    /// PostgreSQL Possible States.
+    /// </summary>
+    public class PostgreSQLState
```
[the remark this answers is not in the document]
Comment (nmalkapuram) [reply src/VirtualClient/VirtualClient.Actions/PostgreSQL/PostgreSQLExecutor.cs:436]: Yes. The state is used in unit and functional testing.

File: src/VirtualClient/VirtualClient.Main/profiles/PERF-POSTGRESQL.json [hunk 2]
```diff
@@ -0,0 +1,57 @@
...
+      }
+    }
+  ],
+  "Dependencies": [
+    {
+      "Type": "DependencyPackageInstallation",
+      "Parameters": {
+        "Scenario": "InstallPostgresqlWorkloadPackage",
+        "BlobContainer": "packages",
+        "BlobName": "postgresql.2.0.0.zip",
```
[the remark this answers is not in the document]
Comment (nmalkapuram) [reply src/VirtualClient/VirtualClient.Main/profiles/PERF-POSTGRESQL.json:20]: sudo apt-get -y install postgresql-14 installs required binaries. The above mentioned  DependencyPackageInstallation installs few .tcl files and scripts that I created which are required to run server and client tasks. These files take parameters that needs to be updated at run time, so haven't added them as part of scripts in our codebase.

File: src/VirtualClient/VirtualClient.Main/profiles/PERF-POSTGRESQL.json [hunk 3]
```diff
@@ -0,0 +1,57 @@
...
+        "PostgresqlPackage": "Postgreqsql",
+        "PackageName": "postgresql",
+        "Version": 14
+        // "Packages-Apt": "build-essential,libevent-dev,pkg-config,zlib1g-dev,libssl-dev,autoconf,automake,libpcre3-dev",
+        // "Packages-Yum": "zlib-devel,pcre-devel,libevent-devel,openssl-devel,git,gcc-c++,make,autoconf,automake",
+        // "Packages-Dnf": "zlib-devel,pcre-devel,libevent-devel,openssl-devel,git,gcc-c++,make,autoconf,automake"
+      }
+    },
+    {
+      "Type": "HammerDbInstallation",
```
[the remark this answers is not in the document]
Comment (nmalkapuram) [reply src/VirtualClient/VirtualClient.Main/profiles/PERF-POSTGRESQL.json:43]: Yes we need it on both sides. When postgresql is installed we have few config files installed along with it which are required for hammerdb client to work as expected on client side. 
And hammerdb is required on client side to start server side task (Creating database and setup configuration at server side)

**Review by brdeyo — CHANGES_REQUESTED** [review]
Hopefully the comments will show up. Apologies as I did not understand that I needed to take an extra step here to submit my comments.

**Comment by nmalkapuram** [comment]
> Hopefully the comments will show up. Apologies as I did not understand that I needed to take an extra step here to submit my comments.

Got them. Thank you

File: src/VirtualClient/VirtualClient.Main/profiles/PERF-SQL-POSTGRESQL.json [hunk 4]
```diff
@@ -0,0 +1,54 @@
+{
+  "Description": "Postgresql Benchmark",
+  "Parameters": {
```
Comment (brdeyo) [inline src/VirtualClient/VirtualClient.Main/profiles/PERF-SQL-POSTGRESQL.json:3]: Please take a look at one of the other profiles that exist in source at the moment. We are starting to add 'Metadata' to the profiles to make them more and more self-describing. As of now we are including 2 or 3 metadata properties that you can see in the other profiles. Please integrate similar ones into this profile when you have a sec. Thanks!

**Review by brdeyo — APPROVED** [review]

**Pull request merged**